mantidproject/mantid
# Fit Script Generator - Create Basic Fit Options Browser

**Description of work.**
This PR creates a Basic Fit Options Browser which provides options for only the most basic fitting options such as `Fitting Mode`, `Max Iterations`, `Minimizer`, `Cost Function` and `Evaluation Type`. This Basic Fit Options Browser will be used in the Fit Script Generator interface.

It also adds a test for this new browser.

**To test:**
1. Comment out this line
https://github.com/mantidproject/mantid/blob/44daa7417226a56f8d1896ea6b2f2b1bc576a5d1/scripts/Muon/GUI/Common/fitting_tab_widget/fitting_tab_view.py#L52
2. Go to Muon Analysis interface
3. click `Load Current Run`
4. Go to Fitting tab and click `Fit Generator`
5. Click `Add Workspaces`. Select the first workspace and click add.
6. Play with the options in the bottom right. There should be no errors.
![image](https://user-images.githubusercontent.com/40830825/112165419-a3191100-8be6-11eb-86b8-c1a50c87407e.png)

7. Switching between Simultaneous and Sequential Fitting Mode should show and hide a column in the FunctionTree, and a function index in the data table.


*No release notes required as this browser is not yet accessible to users*

Fixes #30918

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
